content-security-policy/embedded-enforcement/embedding_csp-header.https.html

```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Embedded Enforcement: Embedding-CSP header.</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="../support/testharness-helper.sub.js"></script>
+</head>
+<body>
+  <script>
+    var tests = [
+      { "name": "Embedding-CSP is not sent if csp attribute is not set on <iframe>.", 
+          "csp": null, "expected": null,"new_csp": null },
+      { "name": "Send Embedding-CSP when csp attribute of <iframe> is not empty.", 
+          "csp": "script-src 'unsafe-inline'", "expected": "script-src 'unsafe-inline'", "new_csp": null },
+      { "name": "Send Embedding-CSP Header on change of `src` attribute on iframe.", 
+          "csp": "script-src 'unsafe-inline'", "expected": "script-src 'unsafe-inline'", "new_csp": "default-src 'unsafe-inline'"},
+      // TODO(amalika): Validate csp attribute to be a proper policy.
+      // { "name": "Send Embedding-CSP Header on change of `src` attribute on iframe.", 
+      //     "csp": "completely wrong csp", "expected": null, "new_csp": null},
+    ];
+
+    tests.forEach(test => {
+      async_test(t =>  {
+        var url = generateURL(Host.SAME_ORIGIN, PolicyHeader.EMBEDDING_CSP);
+        assert_embedding_csp(t, url, test.csp, test.expected, test.new_csp, null);
+      }, "Test same origin: " + test.name);
+    });
+
+    tests.forEach(test => {
+      async_test(t => {
+        var url = generateURL(Host.SAME_ORIGIN, PolicyHeader.CSP);
+        var new_url = generateURL(Host.SAME_ORIGIN, PolicyHeader.EMBEDDING_CSP);
+        assert_embedding_csp(t, url, test.csp, test.expected, test.new_csp, new_url);
+      }, "Test on change of `src` attribute: " + test.name);
+    });
+
+    tests.forEach(test => {
+      async_test(t => {
+        var url = generateURL(Host.SAME_ORIGIN, PolicyHeader.EMBEDDING_CSP);
+        var redirect_url = generateRedirect(Host.SAME_ORIGIN, url);
+        assert_embedding_csp(t, redirect_url, test.csp, test.expected, test.new_csp, null);
+      }, "Test same origin redirect: " + test.name);
+    });
+
+    tests.forEach(test => {
+      async_test(t => {
+        var url = generateURL(Host.SAME_ORIGIN, PolicyHeader.EMBEDDING_CSP);
+        var redirect_url = generateRedirect(Host.CROSS_ORIGIN, url);
+        assert_embedding_csp(t, redirect_url, test.csp, test.expected, test.new_csp, null);
+      }, "Test cross origin redirect: " + test.name);
+    });
+
+    // tests.forEach(test => {
+    //   async_test(t => {
+    //     var url = generateURL(Host.CROSS_ORIGIN, PolicyHeader.EMBEDDING_CSP);
+    //     var redirect_url = generateRedirect(Host.CROSS_ORIGIN, url);
+    //     assert_embedding_csp(t, redirect_url, test.csp, test.new_csp, null);
+    //   }, "Test cross origin redirect of cross origin iframe: " + test.name);
+    // });
```

Comment:
It might timeout because you're loading an insecure frame into a secure document (and is blocked by mixed content).